Purchases/Networking/RCBackend.m

```diff
@@ -479,7 +479,9 @@ - (void)handleSubscriberAttributesResultWithStatusCode:(NSInteger)statusCode
 - (NSDictionary *)attributesUserInfoFromResponse:(NSDictionary *)response statusCode:(NSInteger)statusCode {
     NSMutableDictionary *resultDict = [[NSMutableDictionary alloc] init];
     BOOL isInternalServerError = statusCode >= RC_INTERNAL_SERVER_ERROR;
-    resultDict[RCSuccessfullySyncedKey] = @(!isInternalServerError);
+    BOOL isNotFoundError = statusCode == RC_NOT_FOUND_ERROR;
+    BOOL successfullySynced = !(isInternalServerError || isNotFoundError);
+    resultDict[RCSuccessfullySyncedKey] = @(successfullySynced);
```

Comment:
@vegaro we will actually need this changes in 3.6.0 before releasing, otherwise we'll be doing a regression. maybe these could be separated into a different PR?

Reply:
I'm making a PR cherry-picking the changes for the 404 so we can merge it into 3.6.0 before releasing

Reply:
https://github.com/RevenueCat/purchases-ios/pull/328